react-components/src/components/ViewerAnchor/ViewerAnchor.tsx

```diff
@@ -2,46 +2,92 @@
  * Copyright 2023 Cognite AS
  */
 
-import { useEffect, useRef, type ReactElement, type RefObject, useState, useCallback } from 'react';
+import {
+  useEffect,
+  useRef,
+  type ReactElement,
+  type RefObject,
+  useState,
+  useCallback,
+  type CSSProperties
+} from 'react';
 import { Vector2, type Vector3 } from 'three';
 
 import { useReveal } from '../RevealContainer/RevealContext';
+import { type Cognite3DViewer } from '@cognite/reveal';
+import { clamp } from 'lodash';
 
-export type ViewerAnchorElementMapping = {
-  ref: RefObject<HTMLElement>;
-  position: Vector3;
-};
+type ViewerAnchorStyle = Omit<CSSProperties, 'position' | 'left' | 'top'>;
 
 export type ViewerAnchorProps = {
   position: Vector3;
   sticky?: boolean;
   stickyMargin?: number;
+  style?: ViewerAnchorStyle;
   children: ReactElement;
 };
 
 export const ViewerAnchor = ({
-  position,
+  position: position3d,
   children,
+  style: inputStyle,
   sticky,
-  stickyMargin
+  stickyMargin: inputStickyMargin
```

Comment:
will be better to assign a default value here like `stickyMargin = 0` than doing it as `const stickyMargin = inputStickyMargin ?? 0`

Reply:
It seems to be a little complicated, because `stickyMargin` is not an individual parameter, but a part of the props-object. So I can't give at it a default value alone.